diff --git a/codex-rs/Cargo.lock b/codex-rs/Cargo.lock
index 471c342ff968bc77396ae2d238f18a83865b484c..0d13acefac624d827c64b8eddbe4ecf923edd72f 100644
--- a/codex-rs/Cargo.lock
+++ b/codex-rs/Cargo.lock
@@ -2838,6 +2838,7 @@ dependencies = [
  "sha1 0.10.6",
  "shlex",
  "similar",
+ "sqlx",
  "tempfile",
  "test-case",
  "test-log",
diff --git a/codex-rs/core/Cargo.toml b/codex-rs/core/Cargo.toml
index ff683fb5f921dcf23fe52ef52db6cf9889ca8efe..7639f43cdc7a439ede4d43dc28bf256845d10ae4 100644
--- a/codex-rs/core/Cargo.toml
+++ b/codex-rs/core/Cargo.toml
@@ -107,6 +107,7 @@ serde_json = { workspace = true }
 sha1 = { workspace = true }
 shlex = { workspace = true }
 similar = { workspace = true }
+sqlx = { workspace = true }
 tempfile = { workspace = true }
 thiserror = { workspace = true }
 tokio = { workspace = true, features = [
diff --git a/codex-rs/core/src/saffron/mod.rs b/codex-rs/core/src/saffron/mod.rs
index f7ddec47e8a3189f0b530bff337f6c066d0a9de1..5614b5b3d5a07ce746e7922a8721375997b0de92 100644
--- a/codex-rs/core/src/saffron/mod.rs
+++ b/codex-rs/core/src/saffron/mod.rs
@@ -8,6 +8,7 @@
 pub(crate) mod await_exec;
 mod goal_edit;
 pub(crate) mod goal_supervisor;
+mod storage;
 
 use crate::session::session::Session;
 use crate::session::turn_context::TurnContext;
